runtime/cudaq/builder/kernel_builder.cpp

```diff
@@ -45,6 +45,43 @@ namespace cudaq::details {
 /// @brief Track unique measurement register names.
 static std::size_t regCounter = 0;
 
+/// @brief Return a code intrinsic for describing a global pointer to
+/// user-provided state vector data.
+static std::string getStateDataGlobalIntrinsic(std::size_t hashValue,
+                                               std::size_t size) {
+  return fmt::format(
+      fmt::runtime(
+          R"#(llvm.mlir.global external local_unnamed_addr @nvqpp.state.{}() {{addr_space = 0 : i32, alignment = 8 : i64, dso_local}} : !llvm.struct<(ptr<struct<(f64,f64)>>, i32)> {{
+    %0 = llvm.mlir.null : !llvm.ptr<struct<(f64,f64)>>
+    %1 = llvm.mlir.constant({} : i32) : i32
+    %2 = llvm.mlir.undef : !llvm.struct<(ptr<struct<(f64,f64)>>, i32)>
+    %3 = llvm.insertvalue %0, %2[0] : !llvm.struct<(ptr<struct<(f64,f64)>>, i32)> 
+    %4 = llvm.insertvalue %1, %3[1] : !llvm.struct<(ptr<struct<(f64,f64)>>, i32)> 
+    llvm.return %4 : !llvm.struct<(ptr<struct<(f64,f64)>>, i32)>
+  }})#"),
+      hashValue, size);
```

Comment:
Can't we add this to the intrinsics in Intrinsics.cpp by making the hash and size arguments passed in the calling context?

Reply:
I tried at first but could not figure out how to fit it in as-is. It would be great to do so though, maybe needs just some small changes.

Reply:
We don't need these if we just create a cc.global, right?